src/docs/user/references/Variables.adoc

```diff
@@ -0,0 +1,408 @@
+ * Generic information can use in their definition job/task variables patterns, pattern replacements can be done at execution time when using task generic information.
+
+Example of generic information definition:
+
+For example:
+[source, xml]
+----
+<task ... >
+    <genericInformation>
+        <info name="ginfo" value="${test}"/>
+    </genericInformation>
+    ...
+</task>
+----
+
+[[_dynamic_variables]]
+
+==== Dynamic Variables
+
+As opposed to the <<_workflow_variables>>, *dynamic variables* are all variables created outside the XML job definition.
+We have mainly two types of dynamic variables :
+
+- <<_system_variables>>, declared by the ProActive scheduller.
+- <<_inherited_variables>>, created after adding values to the *variables* map.
+
+[[_system_variables]]
+===== ProActive system variables
+
+Some variables are explicitly defined by the Scheduler to retrieve runtime information about a job or a task.
+
+Here is the list of ProActive system variables:
+
+.ProActive variables
+[cols="2,5,2"]
+|===
+| *Variable name* | *Description* | *Type*
+| `PA_JOB_ID`
+| The current job ID.
+| String
+
+| `PA_JOB_NAME`
+| The current job name.
+| String
+
+| `PA_TASK_ID`
+| The current task ID.
+| String
+
+| `PA_TASK_NAME`
+| The current task name.
+| String
+
+| `PA_TASK_ITERATION`
+| The current iteration index, when using <<_loop,looping>>, starts at 0.
+| Integer
+
+| `PA_TASK_REPLICATION`
+| The current iteration index, when using a <<_replicate,replication>>, starts at 0.
+| Integer
+
+| `PA_TASK_PROGRESS_FILE`
+| The path to the progress file, used to set the task's progress.
+
+  You can import and use the utility class `org.ow2.proactive.scripting.helper.progress.ProgressFile` to read/write values to the file identified by this variable.
+| String
+
+| `PA_SCHEDULER_HOME`
+| The path to Scheduler home, where the Scheduler or the Node is installed.
+| String
+
+| `PA_NODESFILE`
+| The path to the hostfile when using a <<_multi_node_task,multi nodes task>>.
+| String
+
+| `PA_NODESNUMBER`
+| The number of acquired nodes when using a <<_multi_node_task,multi nodes task>>.
+| Integer
+
+| `PA_USER`
+| The username of the ProActive user who has submitted the job.
+| String
+
+|===
+
+They can be used inside the workflow with the pattern syntax, for example:
+
+[source, xml]
+----
+<task ...>
+    ...
+     <forkEnvironment workingDir="/opt/${PA_JOB_ID}"></forkEnvironment>
+    ...
+</task>
+----
+
+[[_inherited_variables]]
+===== Script variables
+
+In addition to the ability to use variables directly inside the workflow with the pattern syntax, it is also possible to use
+all type of variables inside scripts thanks to the `variables` map.
+
+Inside a script, `variables` is bound to a http://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Map^].
```

Comment:
this is actually wrong, as it depends on the language used (see below)

Reply:
I think it is important to explain here variable propagation (as the concept is used but not previously defined in the paragraph below)

Reply:
In summary, this paragraph should be reorganized, maybe split in sub paragraphs

Reply:
ok